runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCachePartition.java

```diff
@@ -560,7 +576,24 @@ public void writeProduceEntryStart(
         entryInfo.putShort(6 * Long.BYTES + 3 * Integer.BYTES, ackMode.value());
 
         logFile.appendBytes(entryInfo);
-        logFile.appendBytes(key);
+
+        int validated = 0;
+
+        if (key.value() == null)
+        {
+            logFile.appendBytes(key);
+        }
+        else
+        {
+            final ValueConsumer writeValue = (buffer, index, length) ->
+            {
+                Varint32FW newLength = varIntRW.set(length).build();
+                logFile.appendBytes(newLength);
+                logFile.appendBytes(buffer, index, length);
+            };
+            OctetsFW value = key.value();
+            validated = valueValidator.validate(value.buffer(), value.offset(), value.sizeof(), writeValue);
```

Comment:
If validation is unsuccessful, then we probably don't want to be doing the rest of the work below in this method, agree?